internal/pkgmanager/ecosystem.go

```diff
@@ -87,15 +105,30 @@ func (p *PkgManager) DownloadArchive(name, version, directory string) (string, e
 		return "", fmt.Errorf("no url found for package %s, version %s", name, version)
 	}
 
-	fileName := p.archiveFilename(name, version, downloadURL)
+	baseFilename := p.archiveFilename(name, version, downloadURL)
+	if baseFilename == "" {
+		panic("base filename for archive is empty")
+	}
+
+	// final path if renameWithHash is false
+	destPath := filepath.Join(directory, baseFilename)
+	if err := downloadToPath(destPath, downloadURL); err != nil {
+		return "", err
+	}
 
-	archivePath, err := downloadToDirectory(directory, downloadURL, fileName)
+	if !renameWithHash {
+		// nothing more to do
+		return destPath, nil
+	}
 
+	// TODO this puts the hash after the file extension - see if we can put it before
+	pathWithHash, err := utils.RenameWithHash(destPath, baseFilename+"-", "")
 	if err != nil {
-		return "", err
+		// rename failed but don't remove the file, just return unmodified path
+		return destPath, err
 	}
```

Comment:
Does this need to be done in this method?

It might make more sense to call `RenameWithHash` outside the method (in resultstore) rather than pass in a confusing `bool` in the function args.

(It will also be easier to test)

Reply:
Good call, done